src/core/variable_mx.jl

```diff
@@ -0,0 +1,344 @@
+                if n==m
+                    dict_mat_vars[index][n,n] = mat_nm[index]
+                # a lower triangle element apears in the lower triangle
+                # and also in the upper triangle
+                else
+                    dict_mat_vars[index][n,m] = mat_nm[index]
+                    dict_mat_vars[index][m,n] = mat_nm[index]
+                end
+            end
+        end
+    end
+    # create explicit Symmetric matrices, supported by JuMP
+    # cannot do it directly through JuMP because of name generation
+    # out = Dict{T, Array{JuMP.VariableRef, 2}}([x for x in dict_mat_vars])
+    out = convert(Dict{T,Array{JuMP.VariableRef,2}}, dict_mat_vars)
+    return Dict([(k, LinearAlgebra.Symmetric(v)) for (k, v) in out])
+end
+
+
+"""
+Same as variable_mx_real, but adds skew-symmetry structure.
+"""
+function variable_mx_skewsymmetric(model::JuMP.Model, indices::Array{T,1}, N::Int;
+        upper_bound::Union{Missing, Dict{T,<:Array{<:Real,2}}}=missing, lower_bound::Union{Missing, Dict{T,<:Array{<:Real,2}}}=missing,
+        set_diag_to_zero::Bool=true, name="", prefix="") where T
+    # the output is a dictionary of (index, matrix) pairs
+    # the data type of the matrix has to be JuMP.GenericAffExpr, because it
+    # will also contain inverted variables (upper triangle) and potentially
+    # constants (zero) on the diagonal
+    dict_mat_vars = Dict{T, Array{JuMP.GenericAffExpr{Float64,JuMP.VariableRef}, 2}}([(index, zeros(N,N)) for index in indices])
+    for n in 1:N
+        for m in 1:n
+            varname = isempty(prefix) ? "$(n)$(m)_$name" : "$(prefix)_$(n)$(m)_$name"
+            # create the lower triangle element (n,m) for all indices
+            # if diagonal element (n,m) is zero
+            if m==n && set_diag_to_zero
+                mat_nm = Dict{T, Float64}([(index, 0.0) for index in indices])
+            # otherwise, create JuMP variables
+            else
+                mat_nm = _make_matrix_variable_element(model, indices, n, m;
+                    upper_bound=upper_bound, lower_bound=lower_bound, varname=varname)
+            end
+            # unpack element (n,m) to the correct place in the ouput dict
+            for index in indices
+                # a diagonal element only appears on the diagonal
+                if n==m
+                    dict_mat_vars[index][n,n] = mat_nm[index]
+                # a lower triangle element apears in the lower triangle
+                # and also in the upper triangle (but inverted)
+                else
+                    dict_mat_vars[index][n,m] =  mat_nm[index]
+                    dict_mat_vars[index][m,n] = -mat_nm[index]
+                end
+            end
+        end
+    end
+    return dict_mat_vars
+end
+
+
+"""
+Returns a pair of symmetric and skew-symmetric matrix variables.
+"""
+function variable_mx_hermitian(model::JuMP.Model, indices::Array{T,1}, N::Int;
```

Comment:
maybe name should be `variable_mx_real_hermitian` (if i'm reading the function definition right this is real and not complex correct?)

Reply:
It returns the complex matrix variable M as a tuple of two real matrix variables, the real and imaginary part (Mre, Mim).

Is it useful to indicate that the matrix variable is real? All matrix variables are real in the end, complex variables are only implied through the composition of their real components (rectangular or polar).

Reply:
In this case I was only trying to tie functions back to their origins, e.g. in the docstring of `variable_mx_symmetric` you say `"Same as variable_mx_real"`. If that is not the case for this function then I agree the rename would not make sense

Reply:
It returns a tuple of real matrix variables, which represent the real and imaginary component Mre and Mim of the complex (and Hermitian) matrix variable M.

So it is 'implicitly complex', I would not add the qualifier 'real' here.

Reply:
Sorry, I edited my comment as I was reading the comments top-to-bottom and realized what you meant after writing my comment.